src/operator/contrib/psroi_pooling.cc

```diff
@@ -38,25 +38,193 @@ using std::floor;
 using std::ceil;
 
 namespace mshadow {
+
+template <typename DType>
+ inline void PSROIPoolForwardCPU(
+  const int count,
+  const DType* bottom_data,
+  const DType spatial_scale,
+  const int channels,
+  const int height, const int width,
+  const int pooled_height, const int pooled_width,
+  const DType* bottom_rois,
+  const int output_dim,
+  const int group_size,
+  DType* top_data) {
+  const int omp_threads = mxnet::engine::OpenMP::Get()->GetRecommendedOMPThreadCount();
+#pragma omp parallel for num_threads(omp_threads)
+  for (int index = 0; index < count; index++) {
+    // The output is in order (n, ctop, ph, pw)
```

Comment:
Can you please add code documentation for the algorithm. That would be very helpful for future maintenance.